srcpkgs/base-files/files/vkpurge

```diff
@@ -3,112 +3,126 @@
 # A simple script to remove old kernel files/modules.
 # Brought to you by yours truly Juan RP in the Public Domain.
 #
-: ${progname:=$(basename $0)}
+: "${progname:="${0##*/}"}"
 
 case "$(xbps-uhelper arch)" in
     aarch64*|ppc64*) KIMAGE="vmlinux";;
     *) KIMAGE="vmlinuz";;
 esac
 
-usage()
-{
+RUNNING=$(uname -r)
+
+usage() {
 	cat <<_EOF
-Usage: $progname <target> [<version>]
+Usage: $progname <MODE> [<VERSION>]
 
-Targets:
- list			Lists old installed kernels.
- rm <version...|all>	Remove kernel <version> or all old kernels.
+MODE
+ list [<VERSION>]		lists old kernels
+ rm (all | <VERSION>...)	remove old kernels
 
-Example:
+Examples:
 	$ $progname list
-	$ $progname rm 4.9.81_1
+	$ $progname list 4.18.*
+	$ $progname rm all
+	$ $progname rm 4.18.17_1 4.18.17_2
+	$ $progname rm 4.18.17_*
+	$ $progname rm 4.18.*
 _EOF
 	exit 1
 }
 
-list_kernels()
-{
-	local k kpkg installed kver _f
-
-	for k in /boot/$KIMAGE-*; do
-		_f=$(basename $k)
-		kver=$(echo ${_f}|sed -e 's|-rc|rc|')
-		installed=$(xbps-query -o "$k" 2>/dev/null)
-		[ -n "$installed" ] && continue
-		kpkg="$(xbps-uhelper getpkgversion ${_f} 2>/dev/null)"
-		[ "$(uname -r)" = "$kpkg" ] && continue
-		[ -n "$kpkg" ] && echo "$kpkg"
-	done
+list_kernels() {
+    case "$1" in
+		all) pattern="*" ;;
+		*) pattern=$1 ;;
+    esac
+    for k in /boot/"$KIMAGE"-*; do
+        case "$INSTALLED" in
+			*"$k"*) continue ;;
+        esac
+        kver=${k##*-}
+        case "$kver" in
+			"$RUNNING") ;;
```

Comment:
I guess you could add something like `${RUNNING:=$(uname -r)}` to only get it once and if needed.

Reply:
i made INSTALLED= because it always took over 0.5 seconds, uname -r measured with `bash` takes 0.021 on the same system.